source/common/http/http2/codec_impl.cc

```diff
@@ -984,6 +993,9 @@ int ConnectionImpl::onStreamClose(int32_t stream_id, uint32_t error_code) {
         if (error_code == NGHTTP2_REFUSED_STREAM) {
           reason = StreamResetReason::RemoteRefusedStreamReset;
           stream->setDetails(Http2ResponseCodeDetails::get().remote_refused);
+        } else if (error_code == NGHTTP2_CONNECT_ERROR) {
+          reason = StreamResetReason::ConnectError;
+          stream->setDetails(Http2ResponseCodeDetails::get().remote_reset);
```

Comment:
Should we use a different details string here?

Reply:
eh, I think it's not worth the detail here (I only split out the reason because otherwise it was impossible to unit test) but I will restructure the code so we only setDetails with it once as a matter of principle :-)